### DISCUSSION TITLE
Properties of TreemapController and TreemapElement can't be read | react-chartjs-2

### DISCUSSION THREAD

**Discussion by zeJabun** [kurkle/chartjs-chart-treemap#76] [Q&A]
In my react file:
```
import {Chart as ChartJS} from 'chart.js';
import { Chart } from 'react-chartjs-2';
import {TreemapController, TreemapElement} from "chartjs-chart-treemap";


ChartJS.register(TreemapController, TreemapElement)
```

When attempting to register the TreemapController and TreemapElement in the 'react-chartjs-2' package, it leads to :

`TypeError: Cannot read properties of undefined (reading 'prototype')`
```
   9 | 
  10 | 
> 11 | ChartJS.register(TreemapController, TreemapElement)
  12 | 
  13 | interface TreemapProps {
  14 | 
```

I don't have any issues registering the built in controllers/elements from 'chart.js'

Has anyone else encountered this issue?

**Comment by kurkle** [comment]
What version are you using?

**Comment by kurkle** [comment]
Latest version, without a wrapper: https://codesandbox.io/s/blissful-dew-9thw4?file=/src/App.js

I think you'll need to take this to the react-chartjs-2 repository.

**Comment by argaen** [comment]
I just hit this, only when running tests with jest. version ^2.3.0 on chartjs ^4.4.0.

EDIT: I fixed this by writing the following to `setupTests.js`:

```
jest.mock('chart.js', () => ({
  __esModule: true,
  ...jest.requireActual('chart.js'),
  Chart: {
    register: jest.fn(),
  },
}));
```
